tests/std/tests/P0896R4_ranges_alg_search_n/test.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::search_n(borrowed<false>{}, 13, 42)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::search_n(borrowed<true>{}, 42, 13)), ranges::subrange<int*>>);
+
+using P = pair<int, int>;
+
+struct instantiator {
+    static constexpr array<P, 11> pairs = {
+        P{0, 42}, P{1, 42}, P{1, 42}, P{0, 42}, P{1, 42}, P{1, 42}, P{0, 42}, P{1, 42}, P{1, 42}, P{1, 42}, P{0, 42}};
+
+    template <class Fwd>
+    static constexpr void call() {
+        const Fwd range{pairs};
+
+        // defaulted predicate and projections
+        {
+            const auto result = ranges::search_n(range, 2, P{1, 42});
+            STATIC_ASSERT(same_as<decltype(result), const ranges::subrange<ranges::iterator_t<Fwd>>>);
+            assert(result.begin() == ranges::next(range.begin(), 1));
+            assert(result.end() == ranges::next(range.begin(), 3));
+        }
+        {
+            const auto result = ranges::search_n(ranges::begin(range), ranges::end(range), 2, P{1, 42});
+            STATIC_ASSERT(same_as<decltype(result), const ranges::subrange<ranges::iterator_t<Fwd>>>);
+            assert(result.begin() == ranges::next(range.begin(), 1));
+            assert(result.end() == ranges::next(range.begin(), 3));
+        }
+
+        // explicit predicate
+        {
+            const auto result = ranges::search_n(range, 2, P{1, 42}, ranges::equal_to{});
+            STATIC_ASSERT(same_as<decltype(result), const ranges::subrange<ranges::iterator_t<Fwd>>>);
+            assert(result.begin() == ranges::next(range.begin(), 1));
+            assert(result.end() == ranges::next(range.begin(), 3));
+        }
+        {
+            const auto result =
+                ranges::search_n(ranges::begin(range), ranges::end(range), 2, P{1, 42}, ranges::equal_to{});
+            STATIC_ASSERT(same_as<decltype(result), const ranges::subrange<ranges::iterator_t<Fwd>>>);
+            assert(result.begin() == ranges::next(range.begin(), 1));
+            assert(result.end() == ranges::next(range.begin(), 3));
+        }
+
+        // explicit predicate and one projection

```

Comment:
```suggestion
        // explicit predicate and projection
```
Since I believe you can only pass 0 or 1 projections (as opposed to many other algorithms where you could have `_Proj1` and `_Proj2`.